CMakeLists.txt

```diff
@@ -104,6 +104,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wstrict-aliasing")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wold-style-cast")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wsign-compare")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wunknown-pragmas")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wswitch")
```

Comment:
I was about to write that this changes nothing as `-Wswitch` is part of the **basic** `-Wall` warnings, but I guess qTox doesn't have those enabled.. :(

Reply:
Good point :grimacing: 

Enabled a ton stolen from you and beat qTox into compliance. Updated with a huge diff.

Each refactor commit should be non-functional (except for trivial things like using float precision instead of double precision in UI). Functional commits are marked as "fix".